rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -85,12 +89,15 @@ class DebruijnInterpreter[M[_], F[_]](
     def go(res: Application): M[Unit] =
       res match {
         case Some((continuation, dataList, updatedSequenceNumber, peek)) =>
-          if (persistent || peek)
-            List(
-              dispatcher.dispatch(continuation, dataList, updatedSequenceNumber),
+          if (persistent)
+            runInParallel(dispatch(continuation, dataList, updatedSequenceNumber))(
```

Comment:
Factoring out `dispatch(continuation, dataList, updatedSequenceNumber)` would make future maintenance easier: just one expression to change instead of three.

Reply:
Done